### PR TITLE
[Secure JSON RPC] Added support for https_proxy env variable.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR adds support for proxies to the secure JSON RPC tool. This feature can be enabled by setting the https_proxy env variable (similar to this PR: https://github.com/libra/libra/pull/4404)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

None.

### CODE REVIEW

File: secure/json-rpc/src/lib.rs [hunk 1]
```diff
@@ -177,13 +177,22 @@ impl JsonRpcClient {
...
     fn execute_request(&self, method: String, params: Vec<Value>) -> Response {
-        ureq::post(&self.host)
-            .timeout_connect(REQUEST_TIMEOUT)
-            .send_json(
-                json!({"jsonrpc": JSON_RPC_VERSION, "method": method, "params": params, "id": 0}),
-            )
+        let mut request = ureq::post(&self.host)
+            .timeout_connect(REQUEST_TIMEOUT).build();
+
+        if let Ok(proxy) = env::var("https_proxy") {
```
Comment (mgorven) [inline secure/json-rpc/src/lib.rs:185]: Technically this should use `http_proxy` if the URL is HTTP vs HTTPS, but I think this is fine for now.

Comment (JoshLind) [reply secure/json-rpc/src/lib.rs]: Aah. thanks! I've updated the code to support both. What do you think?

Comment (mgorven) [reply secure/json-rpc/src/lib.rs]: Perfect, thanks!

**Comment by JoshLind** [comment]
Thanks, Michael! 😄 @bors-libra r=mgorven

**Pull request closed without merging**